homeassistant/components/honeywell/climate.py

```diff
@@ -111,8 +117,10 @@ def __init__(self, data, device, cool_away_temp, heat_away_temp):
         self._attr_is_aux_heat = device.system_mode == "emheat"
 
         # not all honeywell HVACs support all modes
-        mappings = [v for k, v in HVAC_MODE_TO_HW_MODE.items() if device.raw_ui_data[k]]
-        self._hvac_mode_map = {k: v for d in mappings for k, v in d.items()}
+        mappings_hvac = [
```

Comment:
I'd drop this intermediate container as it's only used here. We can create the dict comprehension `_hvac_mode_map` directly.